scripts/west_commands/sign.py

```
@@ -237,6 +237,264 @@ def sign(self, command, build_dir, build_conf, formats):
+
+        If configuration is valid, returns parameters needed by imgtool sign command.
+
+        :param mcbdtconf: MCUboot configuration reader
+        :type: MCUbootDTConfig
+        :param bin_size: size of application binary; this is size application will take
+                         on flash.
+        :type bin_size: int
+        :param quiet: suppress extra diagnostic information
+        :type quiet: Bool
+        :returns: (flash alignment, application image offset on flash, image size)
+        :rtype: (int, int, int)
+        '''
+
+        code = mcbdtconf.get_code_partition()
+
+        if 'slot0_partition' not in code.labels:
+            log.die("west sign, for MCUboot in swap mode, expects DT chosen"
+                    " zephyr,code-partition to point to node labeled"
+                    " slot0_partition. Currently selected node is labeled"
+                    " " + str(code.labels))
+
+        if 'slot1_partition' in code.labels:
+            log.die("chosen zephyr,code-partition may not have slot0_partition"
+                    " and slot1_partition at the same time")
+
+        (write_alignment, _) = code.flash_parameters()
+
+        if write_alignment == 0:
+            log.die('expected non-zero flash alignment, but got'
+                    f' DT flash device write-block-size {write_alignment} for {code}')
+
+        write_alignment = __class__.mcuboot_check_alignment(write_alignment)
+
+        cp_addr, cp_size = code.get_parameters()
+
+        other = mcbdtconf.get_partition('slot1_partition')
+
+        if other:
+            _, op_size = other.get_parameters()
+            # Always go with smaller partition size, as swap will not be possible
+            if op_size != cp_size:
+                log.wrn("slot0_partition and slot1_partition differ in size:"
+                        f" ${cp_size} vs ${op_size}. Smaller size will be used.")
```

Have seen some people use different slot sizes because of the trailer information stored on one slot only, might be an idea (in future) to check which slot is larger and ignore the error if it's that one